### ISSUE TITLE
The HoloLens camera doesn't work with OpenCV for Unity 2.1.5

### ISSUE THREAD

**Issue by hkawakita** [EnoxSoftware/HoloLensWithOpenCVForUnityExample#6]
Hi,
I'm trying to use the HoloLens camera with OpenCV for Unity 2.1.5 and HoloLensWithOpenCVForUnityExample, but it doesn't work.
I also heard that another person who had used it with OpenCVForUnityUWP_Beta3.unitypackage couldn't use it with OpenCV for Unity 2.1.5. What's the difference between them?

**Comment by EnoxSoftware** [comment]
What does "HoloLens camera" mean?
Unfortunately, VideoCapture (int index) and open (int index) is not supported on UWP Platform.
https://enoxsoftware.com/opencvforunity/documentation/support-modules/

**Comment by hkawakita** [comment]
"HoloLens camera" means normal RGB camera in HoloLens. I mean HoloLensWithOpenCVForUnityExample with OpenCV for Unity 2.1.5 doesn't work because the camera doesn't work. I confirmed that other functions of OpenCV without the camera worked and that HoloLensWithOpenCVForUnityExample with OpenCV for Unity 2.1.5, the USB webcam and PC worked.
Does HoloLensWithOpenCVForUnityExample use VideoCapture (int index) or open (int index)? If so, were they supported before with OpenCVForUnityUWP_Beta3.unitypackage?
If not, I'd like to download OpenCVForUnityUWP_Beta3.unitypackage tentatively.

**Comment by EnoxSoftware** [comment]
Is Capabilities properly set?
![windowsstoreappcapabilities](https://cloud.githubusercontent.com/assets/7920392/24768424/71caa4a6-1b3d-11e7-9f77-7a10c112f006.PNG)

**Comment by hkawakita** [comment]
Sure. I set WebCam checked.
I followed procedures of the following blog.
http://noshipu.hateblo.jp/entry/2017/02/16/203515
(I'm sorry this page is written in Japanese)
And the above "another person" is the blogger. I asked him first, then he suggested asking here because he also confirmed that the example doesn't work with OpenCV for Unity 2.1.5.

**Comment by EnoxSoftware** [comment]
In general, when App runs in Holographic Emulation, WebCam of PC works. Please install and test on HoloLens actual machine.

**Comment by hkawakita** [comment]
I installed and tested on the actual machine. I confirmed that the camera didn't work. For example, in HoloLens ComicFilter Example, I could see just white rectangle. In other examples, I couldn't see anything except for menu buttons.
If you tell me how to investigate the cause, I will try it.

**Comment by noshipu** [comment]
Hi,
Thank you for creating OpenCVForUnity.

I also have same problem.

I cannot set up on OpenCVForUnity2.1.5.
Because OpenCVForUnityUWP_Beta3.zip does not exist.

Did you change the setup procedure on OpenCVForUnity2.1.5?

**Comment by EnoxSoftware** [comment]
@hkawakita @noshipu 
In my environment HoloLensWithOpenCVForUnityExample worked without any problem.
OpenCV for Unity 2.1.5
Unity 5.5.1f1
Visual Studio Community 2015 Update 3
HoloLens OS version 10.0.14393.693
Could you tell me about your test environment?

1. Create New Project. (HoloLensWithOpenCVForUnityExample)
2. Import HoloToolKit-Unity-v1.5.5.0.unitypackage. (excluding "HoloToolKit-Examples" folder)
3. Setup HoloToolKit. (HoloToolKit > Configure > Apply HoloLens Project Setting)
4. Import OpenCVForUnity2.1.5 and Select MenuItem "Tool > OpenCV for Unity > Set Plugin Import Settings".
5. Import HoloLensWithOpenCVForUnityExample.unitypackage.
6. Add the "Assets/HoloLensWithOpenCVForUnityExample/Scenes/*.unity" files to "Scenes In Build" list in "Build Settings" window.
7. Add the "WebCam" to "Publishing Settings > Capabilities" checklist in the "Player Settings" window.
8. Build and Deploy. (See https://developer.microsoft.com/en-us/windows/holographic/holograms_100)

Also,
Please email me your invoice number and I will send OpenCVForUnityUWP_Beta3.unitypackage to you.
store@enoxsoftware.com

**Comment by hkawakita** [comment]
I'll tell you next Monday. Thanks,

**Comment by noshipu** [comment]
@EnoxSoftware 
Thank you. It worked!
I was not doing 4th("Set Plugin Import Settings") step.

**Comment by hkawakita** [comment]
@EnoxSoftware 
I also confirmed that it worked.
I have did "Set Plugin Import Settings" (not always), so I'm not sure why it didn't work. Maybe that was in the wrong order.
By the way, ComicFilterExample worked, but the face recognition in all examples didn't seem to work. Anyway, I'm satisfied because the camera in the HoloLens worked with OpenCV.
And I found another point: my anti-virus software Avast detected and deleted Plugins/WSA/UWP/x86/opencv_rgbd320.dll as the file infected with Win32:Evo-gen virus.
I'm sorry to bother you & Thank you so much!
